.github/workflows/non_maintainer_check.yaml

```
@@ -0,0 +1,79 @@
+name: check for non-maintainers
+
+on:
+  pull_request_target:
+    branches: [ master ]
+    types: [ opened ]
+  issue_comment:
+    types: [ created ]
+
+jobs:
+  greeting:
+    runs-on: ubuntu-24.04
+    permissions:
+      pull-requests: write
+    if: >-
+      (github.event_name == 'pull_request_target') &&
+      (github.event.pull_request.author_association != 'COLLABORATOR') &&
+      (github.event.pull_request.author_association != 'MANNEQUIN') &&
+      (github.event.pull_request.author_association != 'MEMBER') &&
+      (github.event.pull_request.author_association != 'OWNER')
+    steps:
+    - name: Greeting
+      env:
+        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        PR_NUMBER: ${{ github.event.pull_request.number }}
+        TEMP_DIR: ${{ runner.temp }}
+      run: |
+        cat <<EOF > "${TEMP_DIR}/greeting.md"
+        Thank you for your contribution! We look forward to seeing more from you.
+        Please run the `make check` command to ensure your changes are correct.
```

[IMO] 'correct' sounds stronger

```suggestion
        Please run the `make check` command to ensure your changes will pass the CI.
```